### PR TITLE
BFD-1582: Remove spurious coverage contracts

### PR DESCRIPTION
**Pull request by christrotteradhoc**
**JIRA Ticket:**
[BFD-1582](https://jira.cms.gov/browse/BFD-1582)

**User Story or Bug Summary:**

Remove spurious part A contract information from the Coverage resource. This was likely added for debugging/experimentation at one point and was inadvertently committed.

---

### What Does This PR Do?

Removes the spurious part A contract information from the Coverage resource for V1 and V2 as well as any code or tests that are only used by the removed code.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java [hunk 1]
```diff
@@ -93,21 +89,6 @@ private static Coverage transformPartA(MetricRegistry metricRegistry, Beneficiar
         .getMedicareCoverageStartDate()
         .ifPresent(value -> TransformerUtilsV2.setPeriodStart(coverage.getPeriod(), value));
 
-    // deh start
```
Comment (bfausett77) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java:96]: This makes sense to remove.  The coverage.contain doesn't even show up in the golden bene data.  Plus, coverage.contract isn't required by CARIN or HL7 US-CORE and this is hard coded anyway.

**Review by mackec1 — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]
LGTM

**Review by bfausett77 — APPROVED** [review]
Looks good!  Nice work!

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 2]
```diff
@@ -1696,19 +1696,6 @@ public static void workAroundHAPIIssue1585(RequestDetails requestDetails) {
     }
   }
 
-  /**
```
[the remark this answers is not in the document]
Comment (christrotteradhoc) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:1699]: we use an overloaded method that takes a Long everywhere else. This was left in place just to support this unnecessary code until we got around to removing it.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java [hunk 3]
```diff
@@ -1277,19 +1277,6 @@ static AdjudicationComponent createAdjudicationWithReason(
     }
   }
 
-  /**
```
[the remark this answers is not in the document]
Comment (christrotteradhoc) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:1280]: we use an overloaded method that takes a Long everywhere else. This was left in place just to support this unnecessary code until we got around to removing it.

**Review by bfausett77 — COMMENTED** [review]
Looks good.  Nice work!

**Pull request merged**